### PR TITLE
Add support for Amadeus.tv

### PR DESCRIPTION
**Pull request by ArnauvGilotra**
### Description:

This pull request adds a new extractor for amadeus.tv, a platform hosting a variety of music and cultural videos. The extractor retrieves video information from the site, including the actual video URL from the sourceVideo field in the site's API response. This allows yt-dlp users to download videos from amadeus.tv, which was previously unsupported. The PR includes the necessary code changes, tests, and complies with yt-dlp's coding conventions.

Fixes: #8155  

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/amadeustv.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                'format_id': 'http-%d' % video['definition'],
+                'width': int_or_none(video.get('width')),
+                'height': int_or_none(video.get('height')),
+                'filesize': int_or_none(video.get('size')),
+                'vcodec': video.get('videoStreamList', [{}])[0].get('codec'),
+                'acodec': video.get('audioStreamList', [{}])[0].get('codec'),
+                'fps': int_or_none(video.get('videoStreamList', [{}])[0].get('fps')),
+                'http_headers': {'Referer': 'http://www.amadeus.tv/'}
+            }
+            formats.append(format_info)
```
Comment (seproDev) [inline yt_dlp/extractor/amadeustv.py:52]: I would write this like so to also extract the source video:
```suggestion
        formats = []
        for video in traverse_obj(video_data, ('videoInfo', ('sourceVideo', ('transcodeList', ...)), {dict})):
            if not url_or_none(video.get('url')):
                continue
            formats.append({
                **traverse_obj(video, {
                    'url': 'url',
                    'format_id': ('definition', {lambda x: f'http-{x or "0"}'}),
                    'width': ('width', {int_or_none}),
                    'height': ('height', {int_or_none}),
                    'filesize': (('totalSize', 'size'), {int_or_none}),
                    'vcodec': ('videoStreamList', 0, 'codec'),
                    'acodec': ('audioStreamList', 0, 'codec'),
                    'fps': ('videoStreamList', 0, 'fps', {float_or_none}),
                }, get_all=False),
                'http_headers': {'Referer': 'http://www.amadeus.tv/'}
            })
```

File: yt_dlp/extractor/amadeustv.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+        return {
+            'id': actual_video_id,
+            'title': traverse_obj(video_data, ('videoInfo', 'basicInfo', 'name'), default=video_id),
+            'thumbnail': traverse_obj(video_data, ('coverInfo', 'coverUrl')),
+            'formats': formats,
+            'duration': int_or_none(traverse_obj(video_data, ('videoInfo', 'sourceVideo', 'duration'))),
+        }
```
Comment (seproDev) [inline yt_dlp/extractor/amadeustv.py:60]: Can be written like so:
```suggestion
        return {
            'id': video_id,
            'display_id': display_id,
            'formats': formats,
            **traverse_obj(video_data, {
                'title': ('videoInfo', 'basicInfo', 'name'),
                'thumbnail': ('coverInfo', 'coverUrl', {url_or_none}),
                'duration': ('videoInfo', 'sourceVideo', ('floatDuration', 'duration'), {float_or_none}),
            }, get_all=False),
            **traverse_obj(nuxt_data, ('item', {
                'title': (('title', 'title_en', 'title_cn'), ),
                'description': (('description', 'description_en', 'description_cn'), ),
                'timestamp': ('date', {parse_iso8601}),
                'view_count': ('view', {int_or_none}),
            }), get_all=False),
        }
```
There is no need to provide a default title as that is handled by core code.
I've also added additional metadata from the hydration data. The test will need to be updated.

File: yt_dlp/extractor/amadeustv.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+from .common import InfoExtractor
+from ..utils import (
+    int_or_none,
+    ExtractorError,
+    traverse_obj,
+)
```
Comment (seproDev) [inline yt_dlp/extractor/amadeustv.py:6]: Sorted and with the additional imports:
```suggestion
    ExtractorError,
    float_or_none,
    int_or_none,
    parse_iso8601,
    url_or_none,
)
from ..utils.traversal import traverse_obj
```

**Review by seproDev — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/amadeustv.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            'id': video_id,
+            'display_id': display_id,
+            'formats': formats,
+            **traverse_obj(video_data, {
+                'title': ('videoInfo', 'basicInfo', 'name'),
+                'thumbnail': ('coverInfo', 'coverUrl', {url_or_none}),
+                'duration': ('videoInfo', 'sourceVideo', ('floatDuration', 'duration'), {float_or_none}),
+            }, get_all=False),
+            **traverse_obj(nuxt_data, ('item', {
+                'title': (('title', 'title_en', 'title_cn'), ),
```
Comment (bashonly) [inline yt_dlp/extractor/amadeustv.py:72]:
```suggestion
                'title': (('title', 'title_en', 'title_cn'), {str}),
```

File: yt_dlp/extractor/amadeustv.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                'description': (('description', 'description_en', 'description_cn'), ),
```
Comment (bashonly) [inline yt_dlp/extractor/amadeustv.py:73]:
```suggestion
                'description': (('description', 'description_en', 'description_cn'), {str}),
```

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/amadeustv.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+                }, get_all=False),
+                'http_headers': {'Referer': 'http://www.amadeus.tv/'},
+            })
+
+        return {
+            'id': video_id,
+            'display_id': display_id,
+            'formats': formats,
+            **traverse_obj(video_data, {
+                'title': ('videoInfo', 'basicInfo', 'name'),
```
Comment (seproDev) [inline yt_dlp/extractor/amadeustv.py:67]:
```suggestion
                'title': ('videoInfo', 'basicInfo', 'name', {str}),
```

**Pull request merged**